xDrip/Scenes/Settings/SettingsUnits/SettingsUnitsInteractor.swift

```diff
@@ -27,7 +27,20 @@ final class SettingsUnitsInteractor: SettingsUnitsBusinessLogic, SettingsUnitsDa
     // MARK: Do something
     
     func doLoad(request: SettingsUnits.Load.Request) {
-        let response = SettingsUnits.Load.Response()
+        let user = User.current
+        
+        let response = SettingsUnits.Load.Response(currentSelectedUnit: user.settings.unit) { [weak self] (index) in
+            self?.handleUnitSelection(index)
+        }
         presenter?.presentLoad(response: response)
     }
+    
+    private func handleUnitSelection(_ index: Int) {
+        guard let unit = GlucoseUnit(rawValue: index) else { return }
+        
+        User.current.settings.updateUnit(unit)
```

Comment:
This is the correct logic, but we also have to propagate settings changes globally in order for other screens could respond to these changes accordingly. There could be a lot of approaches, but let's use the simplest - notifications, but a bit modified. In `BaseSettings` folder create `Settings+Notifications` with the following code (you may modify it if needed - didn't test it):
```swift
extension NotificationCenter {
    enum Setting {
        case unit
    }
    
    func subscribe(forSettingsChange settings: [Setting], notificationHandler: @escaping () -> Void) -> [NSObjectProtocol] {
        let handler: (Notification) -> Void = { _ in
            notificationHandler()
        }
        
        return settings.map { addObserver(forName: $0.notificationName, object: nil, queue: nil, using: handler) }
    }
    
    func postSettingsChangeNotification(setting: Setting) {
        post(name: setting.notificationName, object: nil)
    }
}

fileprivate extension NotificationCenter.Setting {
    var notificationName: Notification.Name {
        let name: String
        switch self {
        case .unit: name = "settingsUnit"
        }
        return Notification.Name(name)
    }
}
```

I didn't put a raw value for Setting enum with purpose, because those names are too short and could interfere with something else.